src/common/btree.c

```diff
@@ -4640,3 +4733,48 @@ dbtree_overhead_get(int alloc_overhead, unsigned int tclass, uint64_t otype,
 	return 0;
 }
 
+int
+dlck_dbtree_check(daos_handle_t toh, struct dlck_print *dp)
+{
+	daos_handle_t ih;
+	int           rc;
+
+	D_ASSERT(dp != NULL);
+
+	DLCK_PRINT(dp, "Nodes:\n");
+	dlck_print_indent_inc(dp);
+
+	rc = dbtree_iter_prepare(toh, 0 /** options */, &ih);
+	if (rc != 0) {
+		DLCK_PRINTF_ERR(dp, "failed to prepare tree iterator: " DF_RC "\n", DP_RC(rc));
+		dlck_print_indent_dec(dp);
+		return rc;
+	}
+
+	rc = dbtree_iter_probe(ih, BTR_PROBE_FIRST, DAOS_INTENT_CHECK, NULL /** key */,
+			       NULL /** anchor */, dp);
+	if (rc == -DER_NONEXIST) {
+		(void)dbtree_iter_finish(ih);
+		dlck_print_indent_dec(dp);
+		return DER_SUCCESS;
+	}
+	if (rc != DER_SUCCESS) {
+		DLCK_PRINTF_ERR(dp, "failed to initialize tree iterator" DF_RC "\n", DP_RC(rc));
+	}
+
+	while (rc != DER_SUCCESS) {
+		rc = dbtree_iter_next(ih);
+		if (rc == -DER_NONEXIST) {
+			rc = 0;
+			break;
+		} else if (rc != DER_SUCCESS) {
+			DLCK_PRINTF_ERR(dp, "failed to move tree iterator: " DF_RC "\n", DP_RC(rc));
+			break;
+		}
```

Comment:
What will be checked for `else` branch? For the caller from `pool_check`, do we further check whether related container can be opened or not?